### DISCUSSION TITLE
Add a demo website

### DISCUSSION THREAD

**Discussion by JonasKruckenberg** [JonasKruckenberg/imagetools#94] [Ideas]
I feel like withe the number of directives already implemented (which is only going to increase with time) it's easy to get lost.
Most directives would also benefit from a visual representation of what they do.

This issue is to keep track of research & progress towards a docs/demo/playground website that shows off vite-imagetools capabilities.

> If you're looking to contribute, this is a great place to start, as I don't have the time right now to create a whole website!<br>
> so get in touch!

**Comment by yamiteru** [comment]
Once the docs are done and the development settles a bit and the library becomes stable I can help with the demo.

**Comment by DannyFeliz** [comment]
I was looking for this

**Comment by JonasKruckenberg** [comment]
That's what I thought, images are inherently visual so a demo page is definitely a priority.
Might be a good idea to go with a simple page having static examples ( vitepress or similar) first and then add interactivity once the wasm integration is a thing

**Comment by yamiteru** [comment]
Should we just create examples folder in this repo or create a new repo `imagetools-examples` (or `imagetools-templates`)? I think the second option is better.

**Comment by yamiteru** [comment]
I wanted to create a demo site, but importing metadata as `import { width, height, src } from "01.jpg"` doesn't work. Importing it like this `import image from "01.jpg"` works. 

EDIT: it works in the next branch so it's alright

**Comment by JonasKruckenberg** [comment]
I've set up the branch next to be a monorepo anyway, so I think it'd be best to just have a folder called examples maybe?
(Similar to how vite does it itself)
I'm not decided though. What would be the benefits of a second repo?

Edit:
I didn't see your last message sorry! Did you add the meta(/metadata) directive?
By default the plugin is exporting the urls to match the same behavior you'd get without it

**Comment by yamiteru** [comment]
I don't know.. I just don't like to clutter my repos. I prefer to create multiple repos. It's just a personal preference.

Yes I'm using the meta directive.

**Comment by DannyFeliz** [comment]
I tend to use the folder `example` or `demo` and there I save all my stuff related to the demo.

**Comment by JonasKruckenberg** [comment]
> I don't know.. I just don't like to clutter my repos. I prefer to create multiple repos. It's just a personal preference.

Yeah that's true and I'd choose a separate repo normally as well, but I think when going down the monorepo route it makes more sense creating an examples folder 🤔

> Yes I'm using the meta directive.

That's odd... Are you using the next version or the current one?
(And maybe we can move the discussion regarding that bug to it's own issue)

**Comment by yamiteru** [comment]
Yes given that this became a monorepo it makes more sense.

It was v3.1.0. It works in the next version so it's alright.

**Comment by cronin4392** [comment]
I'd be interested in helping on this front. I will try to find some time to jam on an idea and get your feedback @JonasKruckenberg!
